client/ayon_core/hosts/max/plugins/publish/validate_no_animation.py

```diff
@@ -0,0 +1,67 @@
+# -*- coding: utf-8 -*-
+import pyblish.api
+from pymxs import runtime as rt
+from ayon_core.pipeline import (
+    PublishValidationError,
+    OptionalPyblishPluginMixin
+)
+from ayon_core.hosts.max.api.action import SelectInvalidAction
+
+
+class ValidateNoAnimation(pyblish.api.InstancePlugin,
+                          OptionalPyblishPluginMixin):
+    """Validates No Animation
+
+    Ensure no keyframes on nodes in the Instance
+    """
+
+    order = pyblish.api.ValidatorOrder
+    families = ["model"]
+    hosts = ["max"]
+    optional = True
+    label = "Validate No Animation"
+    actions = [SelectInvalidAction]
+
+    def process(self, instance):
+        if not self.is_active(instance.data):
+            return
+        invalid = self.get_invalid(instance)
+        if invalid:
+            bullet_point_invalid_statement = "\n".join(
+                "- {}: {}".format(obj, message)
+                for obj, message in invalid
+            )
+            raise PublishValidationError(
+                "Keyframes found on:\n\n{0}".format(
+                    bullet_point_invalid_statement)
+                ,
+                title="Keyframes on model"
+            )
+
+    @staticmethod
+    def get_invalid(instance):
+        invalid = []
+        selected_objects = instance.data["members"]
+        for sel in selected_objects:
+            sel_pos_ctl = rt.getPropertyController(
+                sel.controller, 'Position')
+            ctl_count = (sel_pos_ctl.keys).count
+            if ctl_count > 0:
+                invalid.append(
+                   ( (sel), f"Object Position(s) has {ctl_count} keyframe(s)"))
+            sel_rot_ctl = rt.getPropertyController(
+                sel.controller, "Rotation"
+            )
+            ctl_count = (sel_rot_ctl.keys).count
+            if ctl_count > 0:
+                invalid.append(
+                    ((sel), f"Object Rotation(s) has {ctl_count} keyframe(s)"))
+            sel_scale_ctl = rt.getPropertyController(
+                sel.controller, "Scale"
+            )
+            ctl_count = (sel_scale_ctl.keys).count
+            if ctl_count > 0:
+                invalid.append(
+                    ((sel), f"Object Scale(s) has {ctl_count} keyframe(s)"))
```

Comment:
Shouldn't we somehow just check whether it has ANY animation instead of only checking pos, rot, scale? Isn't there an equivalent "check if it has any keys?" type logic?

Also, why is `sel` between brackets like `(sel)` that seems redundant?

Reply:
this is perhaps one of the ways check whether there is keyframe in Max. But from my current experience I can say they are not Maya alike.

Reply:
You should at least be able to do:
```
numKeys controller
```

According to:
 https://help.autodesk.com/view/MAXDEV/2023/ENU/?guid=GUID-B1700B1D-B1EA-4A6C-B4A3-A29DB26C8C02